src/interfaces/IXPProvider.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+interface IXPProvider {
+    function getTotalXPContribution() external view returns (uint256);
+    function getUserXPContribution(address user) external view returns (uint256);
```

Comment:
This is just nitpicking at this point, but how do you feel about `getTotalXP` and `getUserXPShare` instead?